### PR TITLE
More tests for edge-cases and better coverage

### PR DESCRIPTION
**Pull request by jbt**
I'd also like to get istanbul on here properly - it's from coverage checks that I discovered the missing full-ipv6-address parsing test.

### CODE REVIEW

File: test/reader-test.js [hunk 1]
```diff
@@ -127,6 +130,52 @@ function checkIPv6(reader){
...
+    it('throws with bad search tree', function(){
+      assert.throws(function(){
+        reader.lookup('1.1.1.32');
+      });
+    });
+
+    // I can't figure out why this doesn't throw. I _think_
+    // it's supposed to test whether a pointer can point to
+    // another pointer. Problem is, we don't throw on that.
+    // it('throws with bad data section', function(){
```
Comment (oschwald) [inline test/reader-test.js]: I am not sure if it is related, but this test database was broken upstream for a while. It has since [been fixed](https://github.com/maxmind/MaxMind-DB/pull/15).

Comment (jbt) [inline test/reader-test.js]: Ah great, I'll grab the latest and see if it works for that :)

Comment (jbt) [inline test/reader-test.js]: Yup, updated version works and throws properly. Thanks!

**Comment by jbt** [comment]
Right I think this looks good for now. I'll put something separate together for using istanbul for proper coverage.

Notable things missing from coverage at the moment:

 * Payload size with value = 30 or 31 (i.e. 285 + two-bytes and 65821 + three-bytes values)
 * Pointers with pointer size = 3
 * Uint16 with length = 2
 * Metadata with a bad record size (ie not 24, 28 or 32)

Will need to figure out how to make test-cases for these. I also know the Uint16 / Int16 test isn't good enough because it was passing even when we were reading everything as unsigned.

**Comment by jbt** [comment]
Actually on that note I'm pretty sure pointers with size = 3 (i.e. read as 32-bit value) are actually technically broken because they read as signed rather than unsigned.
